modules/fundamental/src/vulnerability/service/mod.rs

```diff
@@ -155,6 +162,88 @@ impl VulnerabilityService {
 
         Ok(result.rows_affected)
     }
+
+    pub async fn analyze_purls<C: ConnectionTrait>(
+        &self,
+        purls: Vec<&str>,
+        connection: &C,
+    ) -> Result<HashMap<String, Vec<VulnerabilityDetails>>, Error> {
+        if purls.is_empty() {
+            return Ok(Default::default());
+        }
+
+        let query = purls.iter().filter_map(|p| {
+          let purl = Purl::from_str(p).map_err(Error::Purl);
+          let canonical = match purl {
+            Ok(purl) =>{
+              CanonicalPurl::from(purl)
+            },
+            Err(_) => return None
+          };
+
+          let version =  match canonical.version.as_ref() {
+            Some(v) => {
+              v
+            }
+            None => return None
+          };
+
+          let name = canonical.name;
+          let ty = canonical.ty;
+          let ns_condition = match canonical.namespace.clone() {
+            Some(namespace) => format!(r#"base_purl.namespace = '{namespace}'"#),
+            None => String::from("base_purl.namespace is null")
+          };
+
+          Some(format!(r#"
```

Comment:
We should try to avoid using string injection of values, due to obvious issues with SQL injection stuff.

You can use database level parameters like this: https://github.com/trustification/trustify/blob/5e1075324e8821c8a636725c7da004127882cd2f/modules/analysis/src/service/load.rs#L167-L168

Reply:
You're right. Fixed.